### PR TITLE
Updates to MathOptFormat v0.4

### PR DESCRIPTION
**Pull request by odow**
Main changes:

- `deps/build.jl` script is removed in favor of adding the schema file to `src/MOF`
- handling of multiple objectives has changed as per changes to MOF schema
- OrderedCollections used instead of DataStructures
- Compat bounds for all packages
- Small re-organization of compression schemes

cc @dourouc05

### CODE REVIEW

File: src/MOF/read.jl [hunk 1]
```diff
@@ -7,13 +7,14 @@ function MOI.read_from_file(model::Model, io::IO)
         validate(io)
     end
     object = JSON.parse(io; dicttype=Object)
-    if object["version"] > VERSION
+    file_version = _parse_mof_version(object["version"])
+    if file_version > VERSION
         error("Sorry, the file $(filename) can't be read because this library" *
-              " supports version $(VERSION) of MathOptFormat, but the file " *
-              "you are trying to read is version $(object["version"]).")
+              " supports v$(VERSION) of MathOptFormat, but the file you are " *
```
Comment (dourouc05) [inline src/MOF/read.jl]: Minor thing: spacing not consistent within the quotes (line 12: no space at the end of the line; line 13: space).

File: src/MathOptFormat.jl [hunk 2]
```diff
@@ -137,63 +136,88 @@ function create_unique_variable_names(
...
-const _file_formats = Dict{FileFormat, Tuple{String, Any}}(
+const _FILE_FORMATS = Dict{FileFormat, Tuple{String, Any}}(
     # ENUMERATED VALUE => extension, model type
     FORMAT_CBF => (".cbf", CBF.Model),
     FORMAT_LP => (".lp", LP.Model),
     FORMAT_MOF => (".mof.json", MOF.Model),
     FORMAT_MPS => (".mps", MPS.Model)
 )
 
-function _filename_to_format(filename::String)
```
Comment (dourouc05) [inline src/MathOptFormat.jl]: This function is used in the PR on JuMP's side: https://github.com/JuliaOpt/JuMP.jl/pull/1982/files#diff-8d49312dff85b4e2f1ab3e617c3e74e3R24

File: src/MOF/read.jl (hunk 1 above)
Comment (odow) [reply src/MOF/read.jl]: I don't have a consistent rule for this. Something to think about.

File: src/MathOptFormat.jl (hunk 2 above)
Comment (odow) [reply src/MathOptFormat.jl]: It can be re-defined on the JuMP side.

**Pull request merged**